pkg/utils/util.go

```diff
@@ -149,6 +152,33 @@ func ConvertResource(raw []byte, group, version, kind, namespace string) (unstru
 	return *obj, nil
 }
 
+func NormalizeSecret(resource *unstructured.Unstructured) (unstructured.Unstructured, error) {
+	var secret corev1.Secret
+
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(resource.Object, &secret)
```

Comment:
Can we avoid using this converter? I remember this results in a memory leak before.

Reply:
ok, let me check